tools-v2/pkg/cli/command/curvebs/status/volume/snapshot/snapshot.go

```diff
@@ -0,0 +1,204 @@
+/*
+ *  Copyright (c) 2023 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveCli
+ * Created Date: 2023-09-13
+ * Author: saicaca
+ */
+
+package snapshot
+
+import (
+	"encoding/json"
+	"fmt"
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/spf13/cobra"
+	"strconv"
+	"time"
+)
+
+type QueryResponse struct {
+	Code       string              `json:"Code"`
+	Message    string              `json:"Message"`
+	RequestId  string              `json:"RequestId"`
+	TotalCount int                 `json:"TotalCount"`
+	Snapshots  []*SnapshotInfoData `json:"Snapshots,omitempty"`
+}
+
+type SnapshotInfoData struct {
+	UUID         string  `json:"UUID"`
+	User         string  `json:"User"`
+	FileName     string  `json:"FileName"`
+	SnapshotName string  `json:"SnapshotName"`
+	SeqNum       uint64  `json:"SeqNum"`
+	ChunkSize    uint32  `json:"ChunkSize"`
+	SegmentSize  uint64  `json:"SegmentSize"`
+	FileLength   uint64  `json:"FileLength"`
+	Time         uint64  `json:"Time"`
+	Status       int32   `json:"Status"`
+	StripeUnit   *uint64 `json:"StripeUnit,omitempty"`
+	StripeCount  *uint64 `json:"StripeCount,omitempty"`
+	Poolset      *string `json:"Poolset,omitempty"`
+}
+
+const (
+	SNAPSHOT_QUERY_LIMIT          = 100
+	SNAPSHOT_QUERY_INITIAL_OFFSET = 0
+	SNAPSHOT_QUERY_CODE_SUCCESS   = "0"
+	LENGTH_EMPTY                  = 0
+)
+
+const (
+	STATUS_DONE = iota
+	STATUS_IN_PROGRESS
+	STATUS_DELETING
+	STATUS_ERROR_DELETING
+	STATUS_CANCELING
+	STATUS_ERROR
+)
+
+var statusList = []int{
+	STATUS_DONE,
+	STATUS_IN_PROGRESS,
+	STATUS_DELETING,
+	STATUS_ERROR_DELETING,
+	STATUS_CANCELING,
+	STATUS_ERROR,
+}
+
+var statusName = map[int]string{
+	STATUS_DONE:           "done",
+	STATUS_IN_PROGRESS:    "in-progress",
+	STATUS_DELETING:       "deleting",
+	STATUS_ERROR_DELETING: "errorDeleting",
+	STATUS_CANCELING:      "canceling",
+	STATUS_ERROR:          "error",
+}
+
+type SnapshotCommand struct {
+	basecmd.FinalCurveCmd
+
+	snapshotAddrs []string
+	timeout       time.Duration
+	user          string
+	filename      string
```

Comment:
It means that we can query one volume snapshot status? can you give a exmaple in README.md.

And i think it may be error.

Reply:
This argument also exists in the old version tools.
https://github.com/opencurve/curve/blob/ba010be946990b96c42719c39f2d456f9e225aa3/tools/snaptool/querysnap.py#L32-L33

According to the implementation of the backend service, only the snapshots with the matching filename will be queried. It seems that the filename doesn't need to be unique so it can return multiple snapshots.

https://github.com/opencurve/curve/blob/ba010be946990b96c42719c39f2d456f9e225aa3/src/snapshotcloneserver/common/snapshotclone_meta_store_etcd.cpp#L149-L163

The filename is set when creating the snapshot. (#2772) (It is the same in the older version tools)

```bash
curve bs create volume snapshot --user root --filename test --snapshotname snap-test
```

So, I think this argument can be used like

```bash
curve bs status volume snapshot --filename test --snapshotaddr 127.0.0.1:5550,127.0.0.1:5551,127.0.0.1:5552
```

I will add it to the README.